README.md

```diff
@@ -304,6 +304,8 @@ If you're already logged in through any of these methods, the Azure MCP Server w
 
 If you're running into any issues with authentication, visit our [troubleshooting guide](https://github.com/Azure/azure-mcp/blob/main/TROUBLESHOOTING.md).
 
+For advanced authentication scenarios including protected resources, firewall restrictions, and enterprise environments, see our [detailed Authentication guide](https://github.com/Azure/azure-mcp/blob/main/docs/Authentication.md).
```

Comment:
Let's fix the `summary` on line 291 now that this section has expanded to cover so many new details.  Come up with a new concise `summary` tag for this `details` section.